.travis.yml

```diff
@@ -79,6 +79,7 @@ matrix:
         - ES_VERSION=7.3.1
         - TASK=rpkg
       after_success:
+        - .ci/lint.sh
```

Comment:
Can you please . add this logic to `.ci/test.sh` instead of introducing a new file? I would like linting issues to block PRs (i.e. cause a build error), not just get run as an `after_success` step.

Thank you for sharing the link to [Jim Hester's recommended method for using `lintr`](https://github.com/jimhester/lintr#project-configuration) but in my opinion the fact that it spreads the linting details across so many files means it will be harder to maintain and change.

I got similar feedback [on LightGBM](https://github.com/microsoft/LightGBM/pull/2437#discussion_r327587361) and was convinced. Instead of `.lintr` and `lint.sh`, could you just add a `.ci/lint.R` (or whatever you want to call the file) and then call it with `Rscript` from `.ci/test.sh`?